calcom/cal.com
# test: Create 5 tests cases for POST /api/event-types endpoint

## What does this PR do?



- Add 5 tests cases for POST /api/evet-types endpoint:
  - Test for 401 error when non-admin user tries to create an event type with userId
  - Test for 401 error when user cannot access teamId with required roles
  - Test for 400 error when system-wide admin but neither userId nor teamId provided
  - Test for calling checkParentEventOwnership and checkUserMembership if parentId is present
  - Test for successful event type creation when all conditions are met



## Mandatory Tasks (DO NOT REMOVE)


## How should this be tested?



- Environment Variables: Ensure all necessary environment variables for running the tests are correctly configured.
- Minimal Test Data: Use the minimal test data provided in the test files to validate the scenarios.
- **Happy-Path**:
  - Errors:
       ✓ should throw `401` if a non-admin user tries to create an event type with userId
       ✓ should throw `401` if not system-wide admin and user cannot access teamId with required roles
       ✓ should throw `400` if system-wide admin but neither userId nor teamId provided
  - Success:
       ✓ should call `checkParentEventOwnership` and `checkUserMembership` if `parentId` is present
       ✓ should create event type successfully if all conditions are met
- Ensure all input and output scenarios are covered as described in the api/v1/test/lib/event-types/_post.test.ts file.
- You can try run only the test file with `yarn vitest run apps/api/v1/test/lib/event-types/_post.test.ts` or the entire api test command `yarn test api`
- The test result should be showed like bellow:
![Test Result](https://github.com/user-attachments/assets/5552874d-aa0c-4da0-9a3c-979a44fa166e)
